tools/integration-test/src/tests/ics31.rs

```diff
@@ -0,0 +1,159 @@
+            .get_mut("app_state")
+            .and_then(|app_state| app_state.get_mut("epochs"))
+            .and_then(|epochs| epochs.get_mut("epochs"))
+            .and_then(|epochs_list| epochs_list.as_array_mut())
+        {
+            for v in epochs_list {
+                let identifier = v
+                    .get("identifier")
+                    .ok_or_else(|| eyre!("failed to find identifier"))?;
+
+                if identifier.as_str() == Some("stride_epoch") {
+                    let duration = v
+                        .get_mut("duration")
+                        .ok_or_else(|| eyre!("failed to get duration"))?;
+
+                    *duration = serde_json::Value::String("20s".to_owned());
+                }
+            }
+            set_voting_period(genesis, "20s")?;
+            set_staking_max_entries(genesis, "10")?;
+            set_staking_bond_denom(genesis, "stake")?;
+            set_mint_mint_denom(genesis, "stake")?;
+            set_crisis_denom(genesis, "stake")?;
+        }
+        Ok(())
+    }
+
+    // When calling `strided tx stakeibc register-host-zone` new channel
+    // will be created. So the channel worker needs to be enabled.
+    fn modify_relayer_config(&self, config: &mut Config) {
+        config.mode = ModeConfig {
+            connections: config::Connections { enabled: false },
+            channels: config::Channels { enabled: true },
+            ..Default::default()
+        };
+    }
+}
+
+impl BinaryChannelTest for ICS31Test {
+    fn run<ChainA: ChainHandle, ChainB: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        _relayer: RelayerDriver,
+        chains: ConnectedChains<ChainA, ChainB>,
+        channel: ConnectedChannel<ChainA, ChainB>,
+    ) -> Result<(), Error> {
+        let denom_a = chains.node_a.denom();
+        let a_to_b_amount = random_u128_range(1000, 5000);
+        let wallet_a = chains.node_a.wallets().user1().cloned();
+        let wallet_b = chains.node_b.wallets().user1().cloned();
+
+        info!(
+            "Sending IBC transfer from chain {} to chain {} with amount of {} {}",
+            chains.chain_id_a(),
+            chains.chain_id_b(),
+            a_to_b_amount,
+            denom_a
+        );
+
+        chains.node_a.chain_driver().ibc_transfer_token(
+            &channel.port_a.as_ref(),
+            &channel.channel_id_a.as_ref(),
+            &wallet_a.as_ref(),
+            &wallet_b.address(),
+            &denom_a.with_amount(a_to_b_amount).as_ref(),
+        )?;
+
+        let denom_b = derive_ibc_denom(
+            &channel.port_b.as_ref(),
+            &channel.channel_id_b.as_ref(),
+            &denom_a,
+        )?;
+
+        chains.node_b.chain_driver().assert_eventual_wallet_amount(
+            &wallet_b.address(),
+            &denom_b.with_amount(a_to_b_amount).as_ref(),
+        )?;
+
+        info!("Registering host-zone");
+
+        simple_exec(
+            "stride",
+            "strided",
```

Comment:
Does this command always have to be executed with `stride`? Might be better to use the chain driver's command path to execute the command.

Reply:
`gaiad` doesnt have the CLI `<command_path> tx stakeibc register-host-zone`.
I could create move this to a new subdirectory in the `test-framework`:

`test-framework/src/chain/cli/chain-specific/stride/host-zone.rs`

What do you think ?

Reply:
The test itself should not care whether the chain executable supports a given command. If the command is not supported, the test can just fail. We only need to configure the CI matrix such that the supported chain executables are given in the `CHAIN_COMMAND_PATH`.